refactor!: replace onlyOnNoPrefix and onlyOnRoot with redirectOn (#1210)
nuxt-community/nuxt-i18n: docs/content/en/options-reference.md

## `detectBrowserLanguage`

- type: `object`
- default: `{ alwaysRedirect: false, fallbackLocale: '', redirectOn: null, useCookie: true, cookieCrossOrigin: false, cookieDomain: null, cookieKey: 'i18n_redirected', cookieSecure: false }`

Enables browser language detection to automatically redirect visitors to their preferred locale as they visit your site for the first time.

See also [Browser language detection](/browser-language-detection) for a guide.

<alert type="info">

Note that for better SEO it's recommended to set `redirectOn` to `root`.

</alert>

Supported properties:
- `alwaysRedirect` (default: `false`) - Set to always redirect to the value stored in the cookie, not just on first visit.
- `fallbackLocale` (default: `null`) - If none of the locales match the browser's locale, use this one as a fallback.

> - `redirectOn` (default: `null`) - Set to `root` (recommended for improved SEO) to only attempt to detect the browser locale on the root path (`/`) of the site. Only effective when using strategy other than `'no_prefix'`. Set to `no_prefix` as this is a more permissive variant of `root` that will allow attempt to detect the browser locale on the root path (`/`) and also on paths that have no locale prefix (like `/foo`). Only effective when `onlyOnRoot` is not enabled and using strategy other than `'no_prefix'`.

Reviewer: The text flow is not quite right here. It says to set to `root` and then it says to set to `no_prefix`.

And still mentions `onlyOnRoot`.

```suggestion
- `redirectOn` (default: `all`) - Supported options:
  - `all` - detect browser locale on all paths.
  - `root` (recommended for improved SEO) - only detect the browser locale on the root path (`/`) of the site. Only effective when using strategy other than `'no_prefix'`.
  - `no prefix` - a more permissive variant of `root` that will detect the browser locale on the root path (`/`) and also on paths that have no locale prefix (like `/foo`). Only effective when using strategy other than `'no_prefix'`.
```

}
```

> If `true`, a SEO metadata will be generated for the routes. Note that performance can suffer with this option enabled and there might be compatibility issues with some plugins.

Reviewer: This part of `seo` description wasn't removed.

But I've removed all that on `next` branch now since it shouldn't be mixed into this PR.

Author: Sorry, my mistake, I'll remove that now.

Thanks!
